test/e2e/fifo_queue.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
```

Comment:
This queue can be used multiple times because the pop doesn't actually remove an element like the fifo.go implementation does.  This is needed because we continually get the list of pods while checking for the rc to submit the pods, but while that is going on kube is starting pods.  So, we need a fifo that doesn't actually remove elements on a pop.

Alternately we could use a watch to determine total pods and the fifo.go to store pod states.  I'm not sure I like the complexity of adding a watch in addition to the fifo, and it won't help with apiserver stress.  If that is the direction desired, can we merge as is and handle that improvement as a separate pr?